azurelinuxagent/ga/exthandlers.py

```diff
@@ -319,22 +319,13 @@ def handle_ext_handler(self, ext_handler, etag):
 
         try:
             state = ext_handler.properties.state
-            max_retries = 5
-            retry_count = 0
-            while ext_handler_i.decide_version(target_state=state) is None:
-                if retry_count >= max_retries:
-                    err_msg = "Unable to find manifest for extension {0}".format(ext_handler_i.ext_handler.name)
-                    ext_handler_i.set_operation(WALAEventOperation.Download)
-                    ext_handler_i.set_handler_status(message=ustr(err_msg), code=-1)
-                    ext_handler_i.report_event(message=ustr(err_msg), is_success=False)
-                    return
-                time.sleep(2**retry_count * 10)
```

Comment:
Could we perhaps keep 1 (or 2) retries without the sleep?
I am not sure if the agent has any caching for HTTP(S) responses but perhaps clear the cache and retry if the first try fails?

Reply:
The goal state by definition has been updated at this point, so fetching again will not change anything.

Reply:
The goal state contains a URL that points to the extensions config which itself contains a URL for each extension which points to each extension's manifest. I am referring to refetching that extension manifest (e.g. `Publisher.Extension.3.xml` or whatever filename)

Reply:
There was a change in 2.2.27+ where the cache is always cleared when a new goal state is fetched. I believe this addresses your concerns.